File: testdata/typescript/sample.ts
import { useState, useEffect } from 'react';
import type { FC, ReactNode } from 'react';
import * as utils from './utils';
import defaultExport from 'some-module';
import 'side-effect-module';

export interface UserProps {
  name: string;
  age: number;
}

export type Status = 'active' | 'inactive';

export class UserService {
  private baseUrl: string;

  constructor(baseUrl: string) {
    this.baseUrl = baseUrl;
  }

  async getUser(id: number): Promise<UserProps> {
    const response = await fetch(`${this.baseUrl}/users/${id}`);
    return response.json();
  }
}

export function formatName(first: string, last: string): string {
  return `${first} ${last}`;
}

export async function fetchData(url: string): Promise<unknown> {
  const response = await fetch(url);
  return response.json();
}

export const MAX_RETRIES = 3;

export const processItems = (items: string[]) => {
  return items.map(item => item.trim());
};

const internalHelper = (x: number) => x * 2;

function privateFunc(): void {
  console.log('private');
}

export default class App {
  render() {
    return null;
  }
}

File: testdata/python/sample.py
import os
import sys
import json as j
from pathlib import Path
from typing import Optional, List
from collections.abc import Mapping
from ..models import User, Profile
from .utils import helper

MAX_RETRIES = 3
DEFAULT_TIMEOUT = 30

class UserService:
    """Service for managing users."""

    def __init__(self, base_url: str):
        self.base_url = base_url

    def get_user(self, user_id: int) -> Optional[dict]:
        """Fetch a user by ID."""
        pass

    async def update_user(self, user_id: int, data: dict) -> bool:
        """Update a user."""
        pass

    class _InternalHelper:
        """Nested private class."""
        pass


class DataProcessor:
    """Processes data."""

    def process(self, items: List[str]) -> List[str]:
        return [item.strip() for item in items]


def format_name(first: str, last: str) -> str:
    """Format a full name."""
    return f"{first} {last}"


async def fetch_data(url: str) -> dict:
    """Fetch data from a URL."""
    pass


def _private_helper(x: int) -> int:
    """Private helper function."""
    return x * 2
